vivekreddynv/sonic-buildimage
# [RESTAPI] autorestart for restapi process inside the docker is disabled

Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>



#### Why I did it

The critical process in the restapi docker restarts immediately after getting killed even though the FEATURE|restapi|autorestart is set to False.

Critical processes in other dockers follow the same approach.

#### How I did it

#### How to verify it

```
admin@sonic:~$ sudo config feature autorestart restapi disabled 
admin@sonic:~$ docker exec restapi supervisorctl status
dependent-startup                EXITED    Jun 29 12:47 AM
restapi                          RUNNING   pid 21, uptime 0:02:55
rsyslogd                         RUNNING   pid 13, uptime 0:02:57
start                            EXITED    Jun 29 12:47 AM
supervisor-proc-exit-listener    RUNNING   pid 8, uptime 0:02:59

#Kill the Critical Process
admin@sonic:~$ docker exec restapi kill -SIGKILL 21 

admin@sonic:~$ show logging -f | grep restapi
Jun 29 00:51:04.293305 r-lionfish-16 NOTICE restapi#root: message repeated 2 times: [ Waiting for certificates...]
Jun 29 00:51:34.633684 r-lionfish-16 INFO restapi#supervisord 2021-06-29 00:51:34,633 INFO exited: restapi (terminated by SIGKILL; not expected)
Jun 29 00:52:04.295672 r-lionfish-16 INFO restapi#supervisord 2021-06-29 00:52:04,294 INFO reaped unknown pid 67 (exit status 0)
Jun 29 00:52:34.704237 r-lionfish-16 ERR restapi#supervisor-proc-exit-listener: Process 'restapi' is not running in namespace 'host' (1.0 minutes).
Jun 29 00:53:34.769837 r-lionfish-16 ERR restapi#supervisor-proc-exit-listener: Process 'restapi' is not running in namespace 'host' (2.0 minutes).
Jun 29 00:54:34.833258 r-lionfish-16 ERR restapi#supervisor-proc-exit-listener: Process 'restapi' is not running in namespace 'host' (3.0 minutes).
Jun 29 00:55:34.895249 r-lionfish-16 ERR restapi#supervisor-proc-exit-listener: Process 'restapi' is not running in namespace 'host' (4.0 minutes).
```

After the change, the critical process did not restart and it did honor the update made in the FEATURE table. 

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog



#### A picture of a cute animal (not mandatory but encouraged)
